File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        nums.sort()
        if len(nums)==1:
            return nums[0]
        i = 1
        maxm = 1
        maxi = nums[0]
        count = 1
        index = nums[0]
        while i < len(nums):
            if index == nums[i]:
                count = count + 1
                if maxm < count:
                    maxm = count
                    maxi = nums[i]
            else:
                count = 1
                index = nums[i] 
            i = i + 1
        if maxm >= (len(nums)//2):
            return maxi
        else:
            retrun -1
        

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map <int,int> mpp;
        int condition = nums.size()/2;
        for(auto i:nums){
            mpp[i]++;
        }
        for(auto it:mpp){
            if(it.second>condition){
                return it.first;
            }
        }
        return -1;
    }
};